packages/typescript/src/internal/ts_project.bzl

```diff
@@ -0,0 +1,192 @@
+"ts_project rule"
+
+load("@build_bazel_rules_nodejs//:providers.bzl", "DeclarationInfo", "NpmPackageInfo", "run_node")
+
+_ATTRS = {
+    # TODO: limit to the extensions we expect?
+    # ts, tsx, js, jsx, json
+    "srcs": attr.label_list(allow_files = True),
+    "extends": attr.label_list(allow_files = [".json"]),
+    "tsc": attr.label(default = Label("@npm//typescript/bin:tsc"), executable = True, cfg = "host"),
```

Comment:
i'd love to see if this works by replacing this with `ngtsc`

Reply:
yeah I played with it briefly and something didn't work, please give that a try if you have time :)